README.md

```diff
@@ -91,14 +91,35 @@ Results will be added to different tagged releases along with the specific confi
 
 ## Testing
 
-Unit tests can be written for the _reference system_ to check if all nodes, topics and other requirements are met before accepting test results.
+Tests can be written for the _reference system_ to check the platform as well as if all nodes, topics and other requirements are met before accepting test results.
 
 Tests are provided to automatically generate results for you by running `colcon test` on a supported platform above.
 
-To run the test, simply run the following command from your workspace:
+### Dependencies
+
+Before running the tests there are a few prerequisites to complete:
+
+- install LTTNG and `ros2_tracing` [following the instructions in `ros2_tracing`](https://gitlab.com/ros-tracing/ros2_tracing#building)
+   - _Note:_ if you are setting up [ a realtime linux kernel for a raspberry pi using this docker file](https://github.com/ros-realtime/rt-kernel-docker-builder#raspberry-pi-4-rt-linux-kernel), it should [already include LTTNG](https://github.com/ros-realtime/rt-kernel-docker-builder/pull/18)
+- install dependencies using the following command from the `colcon_ws` directory:
+    - `rosdep install --from-paths src --ignore-src -y`
+
+Once the above steps are complete you sould be ready to run the tests and generate some results.
+
+Source your ROS distribution as well as your `ros2_tracing` overlay, compile this repository using the proper CMake arguments and generate some test results:
 
 ```
-colcon build --packages-select autoware_reference_system --cmake-args -DRUN_BENCHMARK=TRUE -DTEST_PLATFORM=TRUE
+# source your ROS distribution
+source /opt/ros/galactic/setup.bash
+# source your ros2_tracing overlay
+source /path/to/ros2_tracing_ws/install/setup.bash
```

Comment:
you might want to try removing this and build `tracetools` / ros2_tracing inside the same workspace used for the reference system

Reply:
that did it. there is now a `ust` folder in the `tracing` directory

Reply:
great!

Reply:
is that expected or a bug of `tracetools` right now?

Reply:
it's not expected, but the issue probably lies in the whole underlay/overlay logic, since it's a linking/library path issue. For all I know this might be an invalid (or flaky/unsupported/weird) overlay setup.

Reply:
still seeing the same `missing events` error though

Reply:
can you share the traces again?

Reply:
`babeltrace **/ust/` shows that there are no `ros2:*` trace events, so yeah something is still wrong. Let me try